src/minmax.jl

```diff
@@ -184,22 +187,34 @@ function maxima(
     idxs = argmaxima(x, w; strict=strict)
     return x[idxs]
 end
+export maxima
 
 """
-    findmaxima(x[, w=1; strict=true]) -> (idxs, vals)
+    findmaxima(x[, w=1; strict=true]) -> NamedTuple
 
-Find the indices and values of local maxima in `x`, where each maxima `i` is either the
-maximum of `x[i-w:i+w]` or the first index of a plateau.
+Find the indices and values of local maxima in `x`, where each maxima `i` is 
+either the maximum of `x[i-w:i+w]` or the first index of a plateau. The 
+returned named tuple contains the fields `indices, heights, data`, 
+and contains what the fieldnames suggest.
```

Comment:
> It might be a good idea to have a short sentence noting that the :data field is a reference of the x argument and not a copy.

Done

As for the other point, it depends on the discussion of the method signature above.